### PR TITLE
fixes traceback issues & adds version feature

### LINKED ISSUE #6: Traceback failure for choice [3] and 7 nodes
CNS metrics backend storage = 4 nodes fails. Inputting 3 nodes works correctly. 

Enter your choice [1-5] : 3
------------------------------------------------------------
For this configuration 7 nodes are recommended
With a minimum of 3 required 
------------------------------------------------------------
How many nodes are available ?:  7
What hosts will be used for application storage (IP/FQDN) ?: aaa.com bbb.com ccc.com
What are the raw storage devices for these hosts(/dev/<device>) ?: /dev/sdb
What is the size of each raw storage device (GB) ?: 100
What is the size for the registry persistent volume (GB)?: 10
What is the size for each metrics persistent volume (GB) ?: 20
What hosts will be used for CNS metrics backend storage  (IP/FQDN) ?:  ddd.com eee.com fff.com ggg.com
What are the raw storage devices for metrics backend on these hosts (/dev/<device>) ?: /dev/sdb
What is the size of each raw storage device (GB) ? : 100
....omitted....
[glusterfs]
aaa.com glusterfs_devices='["/dev/sdb"]'
bbb.com glusterfs_devices='["/dev/sdb"]'
ccc.com glusterfs_devices='["/dev/sdb"]'
Traceback (most recent call last):
  File "./cic.py", line 490, in <module>
    print  app_hosts[3] + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
IndexError: list index out of range

### PR DESCRIPTION
**Pull request by ramkrsna**
This commit fixes #6  and #16 [partially]

### CODE REVIEW

**Comment by netzzer** [comment]
The glusterfs_zone numbering is no longer working correctly.

Wrong in PR #22 
```
[glusterfs]
aaa.com glusterfs_zone=3 glusterfs_devices='["/dev/sdb"]'
bbb.com glusterfs_zone=1 glusterfs_devices='["/dev/sdb"]'
ccc.com glusterfs_zone=3 glusterfs_devices='["/dev/sdb"]'
ddd.com glusterfs_zone=3 glusterfs_devices='["/dev/sdb"]'
```
Correct glusterfs_zone numbering (master branch)
```
[glusterfs]
aaa.com glusterfs_zone=1 glusterfs_devices='["/dev/sdb"]'
bbb.com glusterfs_zone=2 glusterfs_devices='["/dev/sdb"]'
ccc.com glusterfs_zone=3 glusterfs_devices='["/dev/sdb"]'
ddd.com glusterfs_zone=1 glusterfs_devices='["/dev/sdb"]'
```

File: cic.py [hunk 1]
```diff
@@ -136,10 +152,8 @@ def both_in_use():
                 print "   "
 
                 print "[glusterfs]"
-                print  app_hosts[0] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[1] + " glusterfs_zone=2" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[2] + " glusterfs_zone=3" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[3] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
+                for app_host in app_hosts:
+                        print app_host + " glusterfs_zone="+ str(random.randint(1,3)) + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
```
Comment (netzzer) [inline cic.py]: This is not working, numbering should go 1, 2, 3 and then repeat 1, 2, 3 until all nodes have a glusterfs_zone value. Example below for 8 nodes one cluster. 
```
[glusterfs]
aaa.com glusterfs_zone=1 glusterfs_devices='["/dev/sdb"]'
bbb.com glusterfs_zone=2 glusterfs_devices='["/dev/sdb"]'
ccc.com glusterfs_zone=3 glusterfs_devices='["/dev/sdb"]'
ddd.com glusterfs_zone=1 glusterfs_devices='["/dev/sdb"]'
eee.com glusterfs_zone=2 glusterfs_devices='["/dev/sdb"]'
fff.com glusterfs_zone=3 glusterfs_devices='["/dev/sdb"]'
ggg.com glusterfs_zone=1 glusterfs_devices='["/dev/sdb"]'
hhh.com glusterfs_zone=2 glusterfs_devices='["/dev/sdb"]'
```

File: cic.py [hunk 2]
```diff
@@ -160,18 +174,28 @@ def both_in_use():
...
                 print "openshift_storage_glusterfs_storageclass_default=false"
                 print "openshift_storage_glusterfs_block_deploy=false"
                 print "   "
                 print "[glusterfs]"
-                print  app_hosts[0] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[1] + " glusterfs_zone=2" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[2] + " glusterfs_zone=3" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"       
-        
+                for app_host in app_hosts:
+                        print app_host + " glusterfs_zone="+ str(random.randint(1,3)) + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
```
Comment (netzzer) [inline cic.py]: This is not working, numbering should go 1, 2, 3 and then repeat 1, 2, 3.

File: cic.py [hunk 3]
```diff
@@ -309,17 +280,15 @@ def both_in_use():
                 print "   "
                 
                 print "[glusterfs]"
-                print  app_hosts[0] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[1] + " glusterfs_zone=2" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[2] + " glusterfs_zone=3" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print "  "
+                for app_host in app_hosts:
+                        print app_host + " glusterfs_zone="+ str(random.randint(1,3)) + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
```
Comment (netzzer) [inline cic.py]: This is not working, numbering should go 1, 2, 3 and then repeat 1, 2, 3.

File: cic.py [hunk 4]
```diff
@@ -309,17 +280,15 @@ def both_in_use():
...
+                print " "
 
                 print "[glusterfs_registry]"
-                print  log_hosts[0] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(log_devices) + "'"
-                print  log_hosts[1] + " glusterfs_zone=2" + " glusterfs_devices=" + "'" + json.dumps(log_devices) + "'"
-                print  log_hosts[2] + " glusterfs_zone=3" + " glusterfs_devices=" + "'" + json.dumps(log_devices) + "'"
-
-        elif avail_hosts == 4:
+                for log_host in log_hosts:
+                        print log_host + " glusterfs_zone="+ str(random.randint(1,3)) + " glusterfs_devices=" + "'" + json.dumps(log_devices) + "'"
```
Comment (netzzer) [inline cic.py]: This is not working, numbering should go 1, 2, 3 and then repeat 1, 2, 3.

File: cic.py [hunk 5]
```diff
@@ -424,10 +356,9 @@ def both_in_use():
 
                           
                 print "[glusterfs]"
-                print  app_hosts[0] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[1] + " glusterfs_zone=2" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[2] + " glusterfs_zone=3" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-               
+                for app_host in app_hosts:
+                        print app_host + " glusterfs_zone="+ str(random.randint(1,3)) + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
```
Comment (netzzer) [inline cic.py]: This is not working, numbering should go 1, 2, 3 and then repeat 1, 2, 3.

File: cic.py [hunk 6]
```diff
@@ -563,68 +439,14 @@ def both_in_use():
                 print "   "
                 
                 print "[glusterfs]"
-                print  app_hosts[0] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[1] + " glusterfs_zone=2" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[2] + " glusterfs_zone=3" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print " "
+                for app_host in app_hosts:
+                        print app_host + " glusterfs_zone="+ str(random.randint(1,3)) + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
```
Comment (netzzer) [inline cic.py]: This is not working, numbering should go 1, 2, 3 and then repeat 1, 2, 3.

File: cic.py [hunk 7]
```diff
@@ -563,68 +439,14 @@ def both_in_use():
...
-                
-                print "[glusterfs]"
-                print  app_hosts[0] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[1] + " glusterfs_zone=2" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[2] + " glusterfs_zone=3" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[3] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print " "
-
+                for met_host in met_hosts:
+                        print met_host + " glusterfs_zone="+ str(random.randint(1,3)) + " glusterfs_devices=" + "'" + json.dumps(met_devices) + "'"
```
Comment (netzzer) [inline cic.py]: This is not working, numbering should go 1, 2, 3 and then repeat 1, 2, 3.

File: cic.py [hunk 8]
```diff
@@ -676,9 +510,8 @@ def both_in_use():
                           
                 
                 print "[glusterfs]"
-                print  app_hosts[0] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[1] + " glusterfs_zone=2" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[2] + " glusterfs_zone=3" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
+                for app_host in app_hosts:
+                        print app_host + " glusterfs_zone="+ str(random.randint(1,3)) + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
```
Comment (netzzer) [inline cic.py]: This is not working, numbering should go 1, 2, 3 and then repeat 1, 2, 3.

File: cic.py [hunk 9]
```diff
@@ -845,83 +608,15 @@ def both_in_use():
                 print "   "
                 
                 print "[glusterfs]"
-                print  app_hosts[0] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[1] + " glusterfs_zone=2" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[2] + " glusterfs_zone=3" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
+                for app_host in app_hosts:
+                        print app_host + " glusterfs_zone="+ str(random.randint(1,3)) + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
```
Comment (netzzer) [inline cic.py]: This is not working, numbering should go 1, 2, 3 and then repeat 1, 2, 3.

File: cic.py [hunk 10]
```diff
@@ -845,83 +608,15 @@ def both_in_use():
...
                 print "  "
 
                 print "[glusterfs_registry]"
-                print  met_log_hosts[0] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(met_log_devices) + "'"
-                print  met_log_hosts[1] + " glusterfs_zone=2" + " glusterfs_devices=" + "'" + json.dumps(met_log_devices) + "'"
-                print  met_log_hosts[2] + " glusterfs_zone=3" + " glusterfs_devices=" + "'" + json.dumps(met_log_devices) + "'"
+                for met_log_host in met_log_hosts:
+                        print  met_log_host + " glusterfs_zone="+ str(random.randint(1,3)) + " glusterfs_devices=" + "'" + json.dumps(met_log_devices) + "'"
```
Comment (netzzer) [inline cic.py]: This is not working, numbering should go 1, 2, 3 and then repeat 1, 2, 3.

File: cic.py [hunk 11]
```diff
@@ -985,10 +693,9 @@ def both_in_use():
                           
                 
                 print "[glusterfs]"
-                print  app_hosts[0] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[1] + " glusterfs_zone=2" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[2] + " glusterfs_zone=3" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print " "
+                for app_host in app_hosts:
+                        print app_host + " glusterfs_zone="+ str(random.randint(1,3)) + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
```
Comment (netzzer) [inline cic.py]: This is not working, numbering should go 1, 2, 3 and then repeat 1, 2, 3.

File: cic.py [hunk 12]
```diff
@@ -1019,23 +739,37 @@ def both_in_use():
                 print "   "
 
                 print "[glusterfs] "        
-                print  app_hosts[0] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[1] + " glusterfs_zone=2" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[2] + " glusterfs_zone=3" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[3] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
+                for app_host in app_hosts:
+                        print app_host + " glusterfs_zone="+ str(random.randint(1,3)) + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
```
Comment (netzzer) [inline cic.py]: This is not working, numbering should go 1, 2, 3 and then repeat 1, 2, 3.

File: cic.py [hunk 13]
```diff
@@ -1044,10 +778,9 @@ def both_in_use():
                 print "   "
 
                 print  "[glusterfs]"
-                print  app_hosts[0] + " glusterfs_zone=1" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[1] + " glusterfs_zone=2" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-                print  app_hosts[2] + " glusterfs_zone=3" + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
-   
-       
+                for app_host in app_hosts:
+                        print app_host + " glusterfs_zone="+ str(random.randint(1,3)) + " glusterfs_devices=" + "'" + json.dumps(raw_devices) + "'"
```
Comment (netzzer) [inline cic.py]: This is not working, numbering should go 1, 2, 3 and then repeat 1, 2, 3.

**Comment by ramkrsna** [comment]
The glusterfs_zone numbering has been fixed

**Comment by ramkrsna** [comment]
Added the license in 
0552d84 commit.

**Pull request merged**